File: backend/src/main/java/com/example/backend/services/finance/equipment/EquipmentFinanceService.java
package com.example.backend.services.finance.equipment;

import com.example.backend.dto.finance.equipment.EquipmentFinancialSummaryDTO; // ADD THIS
import com.example.backend.models.equipment.Consumable;
import com.example.backend.models.equipment.Equipment;
import com.example.backend.models.warehouse.ItemStatus;
import com.example.backend.repositories.equipment.ConsumableRepository;
import com.example.backend.repositories.equipment.EquipmentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Service
public class EquipmentFinanceService {

    @Autowired
    private EquipmentRepository equipmentRepository;

    @Autowired
    private ConsumableRepository consumableRepository;

    @Transactional
    public void updateEquipmentFinancials(UUID equipmentId) {
        System.out.println("📊 Updating financial tracking for equipment: " + equipmentId);

        Equipment equipment = equipmentRepository.findById(equipmentId)
                .orElseThrow(() -> new IllegalArgumentException("Equipment not found"));

        List<Consumable> inWarehouse = consumableRepository.findByEquipmentIdAndStatus(
                equipmentId, ItemStatus.IN_WAREHOUSE);

        Double currentInventoryValue = inWarehouse.stream()
                .filter(c -> c.getUnitPrice() != null && c.getUnitPrice() > 0)
                .mapToDouble(c -> c.getQuantity() * c.getUnitPrice())
                .sum();

        List<Consumable> consumed = consumableRepository.findByEquipmentIdAndStatus(
                equipmentId, ItemStatus.CONSUMED);

        Double totalExpenses = consumed.stream()
                .filter(c -> c.getUnitPrice() != null && c.getUnitPrice() > 0)
                .mapToDouble(c -> c.getQuantity() * c.getUnitPrice())
                .sum();

        equipment.setCurrentInventoryValue(currentInventoryValue);
        equipment.setTotalExpenses(totalExpenses);
        equipment.setFinanceBalanceUpdatedAt(LocalDateTime.now());

        equipmentRepository.save(equipment);

        System.out.println("✅ Equipment financials updated:");
        System.out.println("   - Current Inventory Value: " + currentInventoryValue + " EGP");
        System.out.println("   - Total Expenses: " + totalExpenses + " EGP");
    }

    public EquipmentFinancialSummaryDTO getEquipmentFinancials(UUID equipmentId) {
        Equipment equipment = equipmentRepository.findById(equipmentId)
                .orElseThrow(() -> new IllegalArgumentException("Equipment not found"));

        updateEquipmentFinancials(equipmentId);

        return EquipmentFinancialSummaryDTO.builder()
                .equipmentId(equipment.getId())
                .equipmentName(equipment.getName())
                .purchasePrice(equipment.getEgpPrice())
                .currentInventoryValue(equipment.getCurrentInventoryValue())
                .totalExpenses(equipment.getTotalExpenses())
                .lastUpdated(equipment.getFinanceBalanceUpdatedAt())
                .build();
    }
}

File: frontend/src/services/finance/equipmentFinanceService.js
import apiClient from '../../utils/apiClient.js';
import { EQUIPMENT_FINANCE_ENDPOINTS } from '../../config/api.config.js';

export const equipmentFinanceService = {
    // ========================================
    // EQUIPMENT FINANCIALS
    // ========================================

    /**
     * Get financial summary for a specific equipment
     * Includes: purchase price, current inventory value, and total expenses
     * @param {string} equipmentId - UUID of the equipment
     * @returns {Promise<Object>} Financial summary with:
     *   - equipmentId
     *   - equipmentName
     *   - purchasePrice (equipment purchase price)
     *   - currentInventoryValue (value of IN_WAREHOUSE consumables)
     *   - totalExpenses (value of CONSUMED consumables)
     *   - lastUpdated
     */
    getEquipmentFinancials: async (equipmentId) => {
        const response = await apiClient.get(
            EQUIPMENT_FINANCE_ENDPOINTS.EQUIPMENT_FINANCIALS(equipmentId)
        );
        return response.data || response;
    },

    /**
     * Manually trigger financial update for an equipment
     * Useful for recalculating values after data corrections
     * @param {string} equipmentId - UUID of the equipment
     */
    updateEquipmentFinancials: async (equipmentId) => {
        const response = await apiClient.post(
            EQUIPMENT_FINANCE_ENDPOINTS.UPDATE_EQUIPMENT_FINANCIALS(equipmentId)
        );
        return response.data || response;
    },
};